### ISSUE TITLE
Switch to use reflection nullability API

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#16062]
Once https://github.com/dotnet/corefx/issues/38087 is done, we should switch to that instead of interpreting [NullableAttribute] ourselves (our method is probably broken by now).

**Comment by ajcvickers** [comment]
@roji Isn't this a duplicate of #15830?

**Comment by roji** [comment]
It most certainly is!
